doc/users/interactive_guide.rst

```diff
@@ -0,0 +1,454 @@
+Conversely, a very slow figure draw will block the prompt until it
+finishes drawing.
+
+Full embedding
+==============
+
+It is also possible to go the other direction and fully embed figures
+(and a `Python interpreter
+<https://docs.python.org/3/extending/embedding.html>`__) in a rich
+native application.  Matplotlib provides classes for each toolkit
+which can be directly embedded in GUI applications (this is how the
+built-in windows are implemented!).  See :ref:`user_interfaces` for
+more details.
+
+
+.. _interactive_scripts :
+
+Scripts and functions
+=====================
+
+
+.. autosummary::
+   :template: autosummary.rst
+   :nosignatures:
+
+   backend_bases.FigureCanvasBase.flush_events
+   backend_bases.FigureCanvasBase.draw_idle
+
+   figure.Figure.ginput
+   pyplot.ginput
+
+   pyplot.show
+   pyplot.pause
+
+There are several use-cases for using interactive figures in scripts:
+
+- capture user input to steer the script
+- progress updates as a long running script progresses
+- streaming updates from a data source
+
+Blocking functions
+------------------
+
+If you only need to collect points in an Axes you can use
+`.figure.Figure.ginput` or more generally the tools from
+`.blocking_input` the tools will take care of starting and stopping
+the event loop for you.  However if you have written some custom event
+handling or are using `.widgets` you will need to manually run the GUI
+event loop using the methods described :ref:`above <cp_block_the_prompt>`.
+
+You can also use the methods described in :ref:`cp_block_the_prompt`
+to suspend run the GUI event loop.  Once the loop exits your code will
+resume.  In general, anyplace you would use `time.sleep` you can use
+`.pyplot.pause` instead with the added benefit of interactive figures.
+
+For example, if you want to poll for data you could use something like ::
+
+  fig, ax = plt.subplots()
+  ln, = ax.plot([], [])
+
+  while True:
+      x, y = get_new_data()
+      ln.set_data(x, y)
+      fig.canvas.draw_idle()
+      plt.pause(1)
+
+which would poll for new data and update the figure at 1Hz.
+
+.. _spin_event_loop:
+
+Explicitly spinning the Event Loop
+----------------------------------
+
+.. autosummary::
+   :template: autosummary.rst
+   :nosignatures:
+
+   backend_bases.FigureCanvasBase.flush_events
+   backend_bases.FigureCanvasBase.draw_idle
+
+
+
+If you have open windows that have pending UI
+events (mouse clicks, button presses, or draws) you can explicitly
+process those events by calling `.FigureCanvasBase.flush_events`.
+This will run the GUI event loop until all UI events currently waiting
+have been processed.  The exact behavior is backend-dependent but
+typically events on all figure are processed and only events waiting
+to be processed (not those added during processing) will be handled.
+
+For example ::
+
+   import time
+   import matplotlib.pyplot as plt
+   import numpy as np
+   plt.ion()
+
+   fig, ax = plt.subplots()
+   fig.canvas.show()
```

Comment:
This doesn't exist, and you have `ion` already.

Reply:
Sorry, marked this resolved, but it isn't.

Reply:
I could have sworn I fixed that and pushed, :sheep:

Reply:
As did I...